File: mobile/android/core/designsystem/src/main/kotlin/com/slopssaloon/omen/core/designsystem/component/OmenMetricStrip.kt
package com.slopssaloon.omen.core.designsystem.component

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import com.slopssaloon.omen.core.designsystem.theme.OmenTheme

/** Sign of the delta drives which invariant token colors the delta text. */
enum class OmenMetricDelta { None, Positive, Negative }

/** One metric row inside a MetricStrip. `confidence` is optional 0..100 for a subline bar. */
data class OmenMetricItem(
    val label: String,
    val value: String,
    val delta: String? = null,
    val deltaDirection: OmenMetricDelta = OmenMetricDelta.None,
    val confidence: Int? = null,
)

/**
 * Registry §3.2 MetricStrip. Labeled metric row(s) with numeric value, optional signed delta,
 * optional confidence subline. Delta color is drawn from the risk invariant family so a
 * positive delta reads as success (risk-low green) and a negative delta as risk (risk-high).
 * Delta always includes a plus/minus glyph in the caller's string so meaning survives
 * grayscale.
 */
@Composable
fun OmenMetricStrip(
    items: List<OmenMetricItem>,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier.fillMaxWidth(),
        verticalArrangement = Arrangement.spacedBy(OmenTheme.spacing.step12),
    ) {
        for (item in items) {
            Column(verticalArrangement = Arrangement.spacedBy(OmenTheme.spacing.step4)) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Text(
                        text = item.label,
                        style = OmenTheme.typography.label.toTextStyle(),
                        color = OmenTheme.color.textSecondary,
                    )
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(OmenTheme.spacing.step8),
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        Text(
                            text = item.value,
                            style = OmenTheme.typography.numeric.toTextStyle(),
                            color = OmenTheme.color.textPrimary,
                        )
                        val deltaText = item.delta
                        if (deltaText != null) {
                            Text(
                                text = deltaText,
                                style = OmenTheme.typography.numeric.toTextStyle(),
                                color = deltaColor(item.deltaDirection),
                            )
                        }
                    }
                }
                val confidence = item.confidence
                if (confidence != null) {
                    OmenConfidenceBar(score = confidence)
                }
            }
        }
    }
}

@Composable
private fun deltaColor(direction: OmenMetricDelta): Color = when (direction) {
    OmenMetricDelta.None -> OmenTheme.color.textSecondary
    OmenMetricDelta.Positive -> OmenTheme.color.data.riskLow
    OmenMetricDelta.Negative -> OmenTheme.color.data.riskHigh
}

File: mobile/ios/OmenIOS/OmenIOS/DesignSystem/OmenMetricStrip.swift
import SwiftUI

/// Sign of the delta drives which invariant token colors the delta text.
enum OmenMetricDelta { case none, positive, negative }

/// One metric row inside a MetricStrip. `confidence` is optional 0..100 for a subline bar.
struct OmenMetricItem: Identifiable {
    let id = UUID()
    let label: String
    let value: String
    let delta: String?
    let deltaDirection: OmenMetricDelta
    let confidence: Int?

    init(
        label: String,
        value: String,
        delta: String? = nil,
        deltaDirection: OmenMetricDelta = .none,
        confidence: Int? = nil
    ) {
        self.label = label
        self.value = value
        self.delta = delta
        self.deltaDirection = deltaDirection
        self.confidence = confidence
    }
}

/// Registry §3.2 MetricStrip. Labeled metric row(s) with numeric value, optional signed
/// delta, optional confidence subline. Delta color is drawn from the risk invariant family
/// so a positive delta reads as success (risk-low green) and a negative delta as risk
/// (risk-high). Delta always includes a plus/minus glyph in the caller's string so meaning
/// survives grayscale.
struct OmenMetricStrip: View {
    let items: [OmenMetricItem]

    var body: some View {
        VStack(alignment: .leading, spacing: OmenSpacing.step12) {
            ForEach(items) { item in
                VStack(alignment: .leading, spacing: OmenSpacing.step4) {
                    HStack {
                        Text(item.label)
                            .omenTextStyle(OmenTypography.label)
                            .foregroundStyle(OmenColor.textSecondary)
                        Spacer(minLength: OmenSpacing.step8)
                        HStack(spacing: OmenSpacing.step8) {
                            Text(item.value)
                                .omenTextStyle(OmenTypography.numeric)
                                .foregroundStyle(OmenColor.textPrimary)
                            if let delta = item.delta {
                                Text(delta)
                                    .omenTextStyle(OmenTypography.numeric)
                                    .foregroundStyle(deltaColor(item.deltaDirection))
                            }
                        }
                    }
                    if let confidence = item.confidence {
                        OmenConfidenceBar(score: confidence)
                    }
                }
            }
        }
    }

    private func deltaColor(_ direction: OmenMetricDelta) -> Color {
        switch direction {
        case .none: return OmenColor.textSecondary
        case .positive: return OmenColor.Data.riskLow
        case .negative: return OmenColor.Data.riskHigh
        }
    }
}

#if DEBUG
#Preview {
    OmenMetricStrip(items: [
        OmenMetricItem(label: "Projected", value: "142.6", delta: "+4.1", deltaDirection: .positive, confidence: 72),
        OmenMetricItem(label: "Opponent", value: "128.4", delta: "−2.3", deltaDirection: .negative),
        OmenMetricItem(label: "Ceiling", value: "168.2"),
    ])
    .padding(OmenSpacing.step16)
    .background(OmenColor.bg)
}
#endif
